### PR TITLE
refactor: Update `README.md` and improved repository

### PR DESCRIPTION
**Pull request by mbfakourii**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

## Issue

Closes: #885

## Approach
Update flutter `README.md`
Update dart `README.md`
Update main `README.md`

Delete unnecessary folders
Fix and add new badges
Fix logo size in [api reference](https://github.com/parse-community/docs/pull/931#issuecomment-1563432423)

## Tasks

- [ ] Add tests
- [x] Add changes to documentation (guides, repository pages, code comments)

### CODE REVIEW

**Comment by Nidal-Bakir** [comment]
So all the docs are now on the [parseplatform](https://parseplatform.org/) site?

**Comment by mbfakourii** [comment]
> So all the docs are now on the [parseplatform](https://parseplatform.org/) site?

yes https://docs.parseplatform.org/flutter/guide/

**Review by Nidal-Bakir — DISMISSED** [review]
Thanks for your work.

File: README.md [hunk 1]
```diff
@@ -4,12 +4,14 @@
 
 [![Build Status](https://github.com/parse-community/Parse-SDK-Flutter/workflows/ci/badge.svg?branch=master)](https://github.com/parse-community/Parse-SDK-Flutter/actions?query=workflow%3Aci+branch%3Amaster)
 [![Coverage](https://img.shields.io/codecov/c/github/parse-community/Parse-SDK-Flutter/master)](https://app.codecov.io/gh/parse-community/Parse-SDK-Flutter/branch/master)
+[![auto-release](https://img.shields.io/badge/%F0%9F%9A%80-auto--release-9e34eb.svg)](https://github.com/parse-community/Parse-SDK-Flutter/releases)
```
Comment (mtrezza) [inline README.md:7]: Please remove, we do not have auto-release here (yet).

File: README.md [hunk 2]
```diff
@@ -21,7 +23,19 @@ These packages are available in this repository:
...
+| Dart    | [parse_server_sdk](./packages/dart)            | [![pub package](https://img.shields.io/pub/v/parse_server_sdk.svg)](https://pub.dev/packages/parse_server_sdk) [![Pub Likes](https://img.shields.io/pub/likes/parse_server_sdk)](https://pub.dev/packages/parse_server_sdk/score) [![Pub Popularity](https://img.shields.io/pub/popularity/parse_server_sdk)](https://pub.dev/packages/parse_server_sdk/score) [![Pub Points](https://img.shields.io/pub/points/parse_server_sdk)](https://pub.dev/packages/parse_server_sdk/score)|
+| Flutter | [parse_server_sdk_flutter](./packages/flutter) | [![pub package](https://img.shields.io/pub/v/parse_server_sdk_flutter.svg)](https://pub.dev/packages/parse_server_sdk_flutter) [![Pub Likes](https://img.shields.io/pub/likes/parse_server_sdk_flutter)](https://pub.dev/packages/parse_server_sdk_flutter/score)  [![Pub Popularity](https://img.shields.io/pub/popularity/parse_server_sdk_flutter)](https://pub.dev/packages/parse_server_sdk_flutter/score) [![Pub Points](https://img.shields.io/pub/points/parse_server_sdk_flutter)](https://pub.dev/packages/parse_server_sdk_flutter/score)|
 
+
+# Guide
+
+See the SDK Usage [guide][guide].
```
Comment (mtrezza) [inline README.md:32]: I'm not sure whether `https://docs.parseplatform.org` is actually in use anywhere. It seems to be an outdated version of the guides list on the front page. On the front page we are actually having 2 separate guides for dart and flutter:
image

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (mbfakourii) [reply README.md]: sure

File: README.md (hunk 2 above)
Comment (mbfakourii) [reply README.md]: I think it would be better to conclude at #931 first

File: packages/dart/README.md [hunk 3]
```diff
@@ -1,903 +1,52 @@
...
+| Version   | Latest Version | End of Support | Compatible                                   |
+|-----------|----------------|----------------|----------------------------------------------|
+| Dart 3.0  | 3.0.0          | May 2024       | ✅ Yes                                        |
+| Dart 2.19 | 2.19.6         | Mar 2024       | ✅ Yes                                        |
+| Dart 2.18 | 2.18.7         | Jan 2024       | ✅ Yes                                        |
+| Dart 2.17 | 2.17.7         | Aug 2023       | ❌ No (Parse Dart SDK requires Dart >=2.18.0) |
+
+## Getting Started
+
+To install, either add [dependency in your pubspec.yaml file](https://pub.dev/packages/parse_server_sdk/install).
```
Comment (mtrezza) [inline packages/dart/README.md:41]:
```suggestion
To install, add the Parse Dart SDK as a [dependency](https://pub.dev/packages/parse_server_sdk/install) in your `pubspec.yaml` file.
```

File: packages/dart/README.md [hunk 4]
```diff
@@ -1,903 +1,52 @@
...
+
+See the SDK Usage [guide][guide].
+
+# Contributing
```
Comment (mtrezza) [inline packages/dart/README.md:45]:
```suggestion
## Contributing

```

File: packages/dart/README.md [hunk 5]
```diff
@@ -1,903 +1,52 @@
...
+| Dart 3.0  | 3.0.0          | May 2024       | ✅ Yes                                        |
+| Dart 2.19 | 2.19.6         | Mar 2024       | ✅ Yes                                        |
+| Dart 2.18 | 2.18.7         | Jan 2024       | ✅ Yes                                        |
+| Dart 2.17 | 2.17.7         | Aug 2023       | ❌ No (Parse Dart SDK requires Dart >=2.18.0) |
+
+## Getting Started
+
+To install, either add [dependency in your pubspec.yaml file](https://pub.dev/packages/parse_server_sdk/install).
+
+See the SDK Usage [guide][guide].
```
Comment (mtrezza) [inline packages/dart/README.md:43]:
```suggestion
## Documentation

Find the full documentation in the [Parse Dart SDK guide][guide].
```

File: packages/dart/README.md [hunk 6]
```diff
@@ -1,903 +1,52 @@
...
+[![Chat](https://img.shields.io/badge/Chat-Join!-%23fff?style=social&logo=slack)](https://chat.parseplatform.org)
+
+---
+
+This library gives you access to the powerful Parse Server backend from your Dart app. For more information on Parse Platform and its features, visit [parseplatform.org](https://parseplatform.org).
+
+---
+
+- [Compatibility](#compatibility)
+- [Getting Started](#getting-started)
```
Comment (mtrezza) [inline packages/dart/README.md:24]:
```suggestion
- [Getting Started](#getting-started)
- [Documentation](#documentation)
- [Contributing](#contributing)
```

File: packages/flutter/README.md [hunk 7]
```diff
@@ -51,974 +36,17 @@ The Parse Flutter SDK is continuously tested with the most recent release of the
...
-    QueryBuilder<ParseObject>(ParseObject('Post'))
-      ..whereValueExists('image', true);
-
-QueryBuilder<ParseObject> queryComment =
-    QueryBuilder<ParseObject>(ParseObject('Comment'))
-      ..whereMatchesQuery('post', queryPost);
-
-var apiResponse = await queryComment.query();
-```
+To install, either add [dependency in your pubspec.yaml file](https://pub.dev/packages/parse_server_sdk_flutter/install).
```
Comment (mtrezza) [inline packages/flutter/README.md:41]:
```suggestion
To install, add the Parse Flutter SDK as a [dependency](https://pub.dev/packages/parse_server_sdk_flutter/install) in your `pubspec.yaml` file.
```

File: packages/flutter/README.md [hunk 8]
```diff
@@ -51,974 +36,17 @@ The Parse Flutter SDK is continuously tested with the most recent release of the
...
 
-If you want to retrieve objects where a field contains an object that does not match another query,  you can use the
-__whereDoesNotMatchQuery__ condition.
-Imagine you have Post class and a Comment class, where each Comment has a pointer to its parent Post.
-You can find comments on posts without images by doing:
+See the SDK Usage [guide][guide].
```
Comment (mtrezza) [inline packages/flutter/README.md:43]:
```suggestion
## Documentation

Find the full documentation in the [Parse Flutter SDK guide][guide].
```

File: packages/flutter/README.md [hunk 9]
```diff
@@ -51,974 +36,17 @@ The Parse Flutter SDK is continuously tested with the most recent release of the
...
 
-```dart
-QueryBuilder<ParseObject> queryPost =
-    QueryBuilder<ParseObject>(ParseObject('Post'))
-      ..whereValueExists('image', true);
+# Contributing
```
Comment (mtrezza) [inline packages/flutter/README.md:45]:
```suggestion
## Contributing
```

File: packages/flutter/README.md [hunk 10]
```diff
@@ -7,36 +21,7 @@ This library gives you access to the powerful Parse Server backend from your Flu
 ---
 
 - [Compatibility](#compatibility)
-- [Example Apps](#example-apps)
 - [Getting Started](#getting-started)
```
Comment (mtrezza) [inline packages/flutter/README.md:24]:
```suggestion
- [Getting Started](#getting-started)
- [Documentation](#documentation)
- [Contributing](#contributing)
```

**Review by mtrezza — CHANGES_REQUESTED** [review]
The package READMEs look good now, very compact.

File: README.md [hunk 11]
```diff
@@ -4,12 +4,13 @@
 
 [![Build Status](https://github.com/parse-community/Parse-SDK-Flutter/workflows/ci/badge.svg?branch=master)](https://github.com/parse-community/Parse-SDK-Flutter/actions?query=workflow%3Aci+branch%3Amaster)
 [![Coverage](https://img.shields.io/codecov/c/github/parse-community/Parse-SDK-Flutter/master)](https://app.codecov.io/gh/parse-community/Parse-SDK-Flutter/branch/master)
+[![License](https://img.shields.io/badge/license-Apache%202.0-lightgrey.svg)](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/LICENSE)
```
[the remark this answers is not in the document]
Comment (mbfakourii) [reply README.md:7]: removed

File: README.md [hunk 12]
```diff
@@ -4,12 +4,13 @@
...
+[![Forum](https://img.shields.io/discourse/https/community.parseplatform.org/topics.svg)](https://community.parseplatform.org/c/parse-server)
```
[the remark this answers is not in the document]
Comment (mbfakourii) [reply README.md:8]: sure

File: packages/dart/README.md [hunk 13]
```diff
@@ -1,903 +1,52 @@
...
-
-Objects:
-* Create new object
-* Extend Parse Object and create local objects that can be saved and retreived
-* Queries
+<img src="https://user-images.githubusercontent.com/5673677/166120960-ea1f58e3-a62b-4770-b541-f64186859339.png" alt="parse-repository-header-sdk-dart" style="max-width: 100%;">
+
+---
+
+[![pub package](https://img.shields.io/pub/v/parse_server_sdk.svg)](https://pub.dev/packages/parse_server_sdk) [![Pub Likes](https://img.shields.io/pub/likes/parse_server_sdk)](https://pub.dev/packages/parse_server_sdk/score) [![Pub Popularity](https://img.shields.io/pub/popularity/parse_server_sdk)](https://pub.dev/packages/parse_server_sdk/score) [![Pub Points](https://img.shields.io/pub/points/parse_server_sdk)](https://pub.dev/packages/parse_server_sdk/score)
```
[the remark this answers is not in the document]
Comment (mbfakourii) [reply packages/dart/README.md:5]: ok

File: packages/flutter/README.md [hunk 14]
```diff
@@ -1,4 +1,18 @@
-![parse-repository-header-sdk-flutter](https://user-images.githubusercontent.com/5673677/166121333-2a144ce3-95bc-45d6-8840-d5b2885f2046.png)
+<img src="https://user-images.githubusercontent.com/5673677/166121333-2a144ce3-95bc-45d6-8840-d5b2885f2046.png" alt="parse-repository-header-sdk-flutter" style="max-width: 100%;">
+
+---
+
+[![pub package](https://img.shields.io/pub/v/parse_server_sdk_flutter.svg)](https://pub.dev/packages/parse_server_sdk_flutter) [![Pub Likes](https://img.shields.io/pub/likes/parse_server_sdk_flutter)](https://pub.dev/packages/parse_server_sdk_flutter/score)  [![Pub Popularity](https://img.shields.io/pub/popularity/parse_server_sdk_flutter)](https://pub.dev/packages/parse_server_sdk_flutter/score) [![Pub Points](https://img.shields.io/pub/points/parse_server_sdk_flutter)](https://pub.dev/packages/parse_server_sdk_flutter/score)
```
[the remark this answers is not in the document]
Comment (mbfakourii) [reply packages/flutter/README.md:5]: ok

**Comment by mbfakourii** [comment]
> The package READMEs look good now, very compact.

changes are applied.
It has a very good structure. 👌👌

Looks like it's ready to merge.

Please re-run the documentation CI after merging to resolve the logo issue.

**Comment by mtrezza** [comment]
image

Could you remove the "docs passing", likes, popularity, points badges? We want to reduce badges in all READMEs and I'm not sure how meaningful they are for developers. They see these infos in the pub.dev registry anyway.

The structure should be:
row #1: technical: image
row #2: versions: image
row #3: marketing image

**Comment by mtrezza** [comment]
Could you add the [Backers](https://github.com/parse-community/parse-server#backers) section before the contributors section in the README?

**Comment by mbfakourii** [comment]
[Done](https://github.com/parse-community/Parse-SDK-Flutter/tree/e9a2d6a9873d2c2e80931a8dbf8e2925eb025338)

**Review by mtrezza — APPROVED** [review]
Looks good, can we merge?

**Comment by mbfakourii** [comment]
> Looks good, can we merge?

Yes, I think it is ready

**Review by mtrezza — APPROVED** [review]

**Comment by mbfakourii** [comment]
@mtrezza 
Please re-run the documentation CI to resolve the logo issue.

**Comment by mtrezza** [comment]
Done

**Comment by mtrezza** [comment]
There was no update, because the docs are created based on a release tag. The tags are in the past, so any changes won’t reflect there yet. The docs will be updated when the next release is published.

**Pull request merged**